doc/rfc-ephemeral-disk-service.md

```diff
@@ -0,0 +1,482 @@
+
+If cloud-init formatted the resource disk with a different mount point, the disk will be in use and will fail to format resulting in service error.
+
+Mitigation: None - exit with failure when format fails.
+
+### Conflict D: Cloud-Init Formats Resource Disk (first), With NVMe, Matching Mount Point
+
+Cloud-init formatted the resource disk with matching mount point.
+
+Mitigation: If target mount point is already mounted, check source device and exit with error if unexpected device.
+
+### Conflict E: Cloud-Init Formats Resource Disk (first), With NVMe, Mismatched Mount Point
+
+Cloud-init formatted the resource disk with different mount point.
+
+This results in an uneccessary mount, and conflict only arises if mounts are dependent and ordered wrong (e.g. /mnt/resource under /mnt).
+
+Mitigation: This falls into the realm of user responsibility, no checks will be performed. Document.
+
+## Cloud-init Workarounds to Proactively Avoid Conflicts
+
+To avoid conflicts with the SCSI resource disk, it would be best to disable cloud-init's configuration of SCSI resource disk.  There is no great way to do this today, but we have some potential workarounds.  If needed, in addition to a long-term solution.
+
+### Cloud-init Workaround 1: Disable resource disk symlinks via udev rules
+
+A udev rules file in `/etc` can be used to mask the cloud-init udev rules, preventing the symlink from being created:
+
+```
+ln -sf /dev/null /etc/udev/rules.d/66-azure-ephemeral.rules
+```
+
+Debugging shows this is well supported by systemd-udevd:
+
+```
+systemd-udevd[202]: Skipping overridden file '/usr/lib/udev/rules.d/66-azure-ephemeral.rules'.
+systemd-udevd[202]: Skipping empty file: /etc/udev/rules.d/66-azure-ephemeral.rules
+```
+
+This is probably the simplest and most effective approach.
+
+### Cloud-init Workaround 2: Redefining ephemeral0 => null
+
+A config can be used to ensure cloud-init never partitions or formats the disk by redefining ephemeral0 such that it doesn't point to `/dev/disk/cloud/azure-resource`.
+
+```yaml
+datasource:
+  Azure:
+    disk_aliases:
+      ephemeral0: null
+
+mounts:
+  - ["ephemeral0.1", null]
+```
+
+**Caveats:**
+
+* this emits a warning when the device isn't found.
+
+### Cloud-init Workaround 3: Re-configure cloud-init mount point
+
+Cloud-init can be reconfigured to mount SCSI resource disk in non-conflicting path.
+
+Example `/etc/cloud/cloud.cfg.d/99-azure-resource.cfg`:
+
+```yaml
+mounts:
+   - ["ephemeral0.1", "/media/azure-scsi-resource"]
+```
+
+**Caveats:**
+
+* Breaks scenario (b) when only SCSI resource disk is available.
+
+### Long-term Plan: First-class Configuration in cloud-init
+
+We will independently work on a toggle for cloud-init to disable SCSI resource disk handling and key off this in a future version.
+
+Example:
+
+```yaml
+datasource:
+  Azure:
+    disable_resource_disk_handling: true
```

Comment:
Stylistic suggestion: Make the key be the affirmative case, and have it default to true. Admin would set it to `false` to suppress disk setup. I'd suggest something like
```yaml
datasource:
   Azure:
      resource_disk_setup: false
```

Reply:
good idea, done!